controlplane/kubeadm/api/v1alpha3/conversion.go

```diff
@@ -17,18 +17,44 @@ limitations under the License.
 package v1alpha3
 
 import (
+	apiconversion "k8s.io/apimachinery/pkg/conversion"
 	"sigs.k8s.io/cluster-api/controlplane/kubeadm/api/v1alpha4"
+
+	utilconversion "sigs.k8s.io/cluster-api/util/conversion"
 	"sigs.k8s.io/controller-runtime/pkg/conversion"
 )
 
 func (src *KubeadmControlPlane) ConvertTo(destRaw conversion.Hub) error {
 	dest := destRaw.(*v1alpha4.KubeadmControlPlane)
-	return Convert_v1alpha3_KubeadmControlPlane_To_v1alpha4_KubeadmControlPlane(src, dest, nil)
+
+	if err := Convert_v1alpha3_KubeadmControlPlane_To_v1alpha4_KubeadmControlPlane(src, dest, nil); err != nil {
+		return err
+	}
+
+	// Manually restore data.
+	restored := &v1alpha4.KubeadmControlPlane{}
+	if ok, err := utilconversion.UnmarshalData(src, restored); err != nil || !ok {
+		return err
+	}
+
+	dest.Spec.RolloutStrategy = restored.Spec.RolloutStrategy
```

Comment:
if we do backport to v1alpha3 wouldn't this be a problem?

Reply:
If we do backport, we need to remove this part of the code and say something like `Required minimum v0.3.x version before upgrading is ...`